### ISSUE TITLE
Issues with version 0.5.2

### ISSUE THREAD

**Issue by ellavas** [phoenixnap/springmvc-raml-plugin#48]
Hi,

I have updated a project to use the version 0.5.2 in order to use the schemaLocation parameter.

Below is one example with pojos, the second one referencing the first one.
sample.json
```json
{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "title": "Sample Schema",
  "id": "Sample",
  "type": "object",
  "description": "The schema of a Sample",
  "properties": {
    "tString": {
      "type": "string",
      "description": "The type string."
    },
    "tInteger": {
      "type": "integer",
      "description": "The type integer."
    },
    "tNumber": {
      "type": "number",
      "description": "The type number.",
      "minimum": 0
    },
    "tDate": {
      "type": "date",
      "description": "The type date."
    },
    "tBoolean": {
      "type": "boolean",
      "description": "The type boolean.",
      "default": true
    },
    "tArrayRef": {
      "type": "array",
      "items": {
        "$ref": "#/definitions/tRef"
      },
      "minItems": 1
    }
  },
  "required": [
    "tInteger",
    "tArrayRef"
  ],
  "definitions": {
    "tRef": {
      "$schema": "http://json-schema.org/draft-04/schema#",
      "title": "tRef",
      "type": "object",
      "description": "A definition reference.",
      "properties": {
        "refId": {
          "type": "string",
          "description": "The identifier of a reference."
        }
      }
    }
  }
}
```
and samples.json which is a collection of sample.json
```json
{
   "$schema": "http://json-schema.org/draft-04/schema",
   "title": "Sample Collection schema",
   "id": "Samples",
   "type": "array",
   "uniqueItems": true,
   "items": {
      "$ref": "sample.json"
   }
}
```

With the following maven configuration
```xml
<plugin>
	<groupId>com.phoenixnap.oss</groupId>
	<artifactId>springmvc-raml-plugin</artifactId>
	<version>0.5.2</version>
	<executions>
		<execution>
			<id>generate-springmvc-endpoints</id>
			<phase>generate-resources</phase>
			<goals>
				<goal>generate-springmvc-endpoints</goal>
			</goals>
			<configuration>
				<ramlPath>src/main/resources/api.raml</ramlPath>
				<rule>com.phoenixnap.oss.ramlapisync.generation.rules.Spring4ControllerInterfaceRule</rule>
				<basePackage>com.test.samples</basePackage>	
				<schemaLocation>src/main/resources/schemas</schemaLocation>
			</configuration>
		</execution>
	</executions>
</plugin>
```

When trying to generate the endpoints, the build success but I got the follwing exception:
```
[ERROR] Error generating pojo from schema Sample
java.lang.IllegalArgumentException: Error parsing document: classpath:/
        at org.jsonschema2pojo.ContentResolver.resolveFromClasspath(ContentResolver.java:88)
        at org.jsonschema2pojo.ContentResolver.resolve(ContentResolver.java:61)
        at org.jsonschema2pojo.SchemaStore.create(SchemaStore.java:48)
        at org.jsonschema2pojo.SchemaStore.create(SchemaStore.java:116)
        at org.jsonschema2pojo.rules.SchemaRule.apply(SchemaRule.java:59)
        at org.jsonschema2pojo.rules.SchemaRule.apply(SchemaRule.java:30)
        at org.jsonschema2pojo.rules.ArrayRule.apply(ArrayRule.java:81)
        at org.jsonschema2pojo.rules.ArrayRule.apply(ArrayRule.java:37)
        at org.jsonschema2pojo.rules.TypeRule.apply(TypeRule.java:107)
        at org.jsonschema2pojo.rules.TypeRule.apply(TypeRule.java:39)
        at org.jsonschema2pojo.rules.SchemaRule.apply(SchemaRule.java:73)
        at org.jsonschema2pojo.rules.SchemaRule.apply(SchemaRule.java:30)
        at org.jsonschema2pojo.rules.PropertyRule.apply(PropertyRule.java:75)
        at org.jsonschema2pojo.rules.PropertyRule.apply(PropertyRule.java:43)
        at org.jsonschema2pojo.rules.PropertiesRule.apply(PropertiesRule.java:64)
        at org.jsonschema2pojo.rules.PropertiesRule.apply(PropertiesRule.java:32)
        at org.jsonschema2pojo.rules.ObjectRule.apply(ObjectRule.java:125)
        at org.jsonschema2pojo.rules.ObjectRule.apply(ObjectRule.java:70)
        at org.jsonschema2pojo.rules.TypeRule.apply(TypeRule.java:89)
        at org.jsonschema2pojo.rules.TypeRule.apply(TypeRule.java:39)
        at org.jsonschema2pojo.rules.SchemaRule.apply(SchemaRule.java:73)
        at org.jsonschema2pojo.rules.SchemaRule.apply(SchemaRule.java:30)
        at org.jsonschema2pojo.SchemaMapper.generate(SchemaMapper.java:115)
        at com.phoenixnap.oss.ramlapisync.naming.SchemaHelper.buildBodyJCodeModel(SchemaHelper.java:427)
        at com.phoenixnap.oss.ramlapisync.naming.SchemaHelper.mapSchemaToPojo(SchemaHelper.java:390)
        at com.phoenixnap.oss.ramlapisync.data.ApiMappingMetadata.parseResponse(ApiMappingMetadata.java:159)
        at com.phoenixnap.oss.ramlapisync.data.ApiMappingMetadata.<init>(ApiMappingMetadata.java:69)
        at com.phoenixnap.oss.ramlapisync.data.ApiMappingMetadata.<init>(ApiMappingMetadata.java:74)
        at com.phoenixnap.oss.ramlapisync.data.ApiControllerMetadata.addApiCall(ApiControllerMetadata.java:56)
        at com.phoenixnap.oss.ramlapisync.generation.RamlParser.checkResource(RamlParser.java:155)
        at com.phoenixnap.oss.ramlapisync.generation.RamlParser.checkResource(RamlParser.java:161)
        at com.phoenixnap.oss.ramlapisync.generation.RamlParser.extractControllers(RamlParser.java:95)
        at com.phoenixnap.oss.ramlapisync.plugin.SpringMvcEndpointGeneratorMojo.generateEndpoints(SpringMvcEndpointGeneratorMojo.java:163)
        at com.phoenixnap.oss.ramlapisync.plugin.SpringMvcEndpointGeneratorMojo.execute(SpringMvcEndpointGeneratorMojo.java:324)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:483)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: com.fasterxml.jackson.core.JsonParseException: Unrecognized token 'sample': was expecting ('true', 'false' or 'null')
 at [Source: java.io.ByteArrayInputStream@149f5761; line: 1, column: 8]
        at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:1581)
        at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:533)
        at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._reportInvalidToken(UTF8StreamJsonParser.java:3451)
        at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._handleUnexpectedValue(UTF8StreamJsonParser.java:2610)
        at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._nextTokenNotInObject(UTF8StreamJsonParser.java:841)
        at com.fasterxml.jackson.core.json.UTF8StreamJsonParser.nextToken(UTF8StreamJsonParser.java:737)
        at com.fasterxml.jackson.databind.ObjectMapper._initForReading(ObjectMapper.java:3776)
        at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:3721)
        at com.fasterxml.jackson.databind.ObjectMapper.readTree(ObjectMapper.java:2211)
        at org.jsonschema2pojo.ContentResolver.resolveFromClasspath(ContentResolver.java:86)
        ... 55 more
```

In addition to that, I have observed that:
- There is no Samples pojo generated.
- The tRef schema defined in sample.json#/definitions/tRef generates a pojo called TArrayRef instead of TRef.

Do I missed something in the configuration or do you have any idea how to solve those issues ?

Thanks a lot

**Comment by kurtpa** [comment]
With this configuration i am unable to generate the exception you listed. The other behaviour however i can replicated and this is because:

1. There is no "Samples" Pojo since you are defining it as an array, not an object. As such it becomes a List<X> where X is the type as defined by the items

2. The tArrayRef schema: From the testing i have done, this is something outside our control and is a behaviour of jsonschema2pojo. I have not found the way to configure this and even going directly to http://www.jsonschema2pojo.org/ produces the same result

**Comment by kurtpa** [comment]
Reference: https://github.com/joelittlejohn/jsonschema2pojo/issues/164

**Comment by ellavas** [comment]
I have psuhed my sample project [here](https://github.com/ellavas/springmvc-raml-plugin-sample) . I hope it will help you to reproduce the exception.

**Comment by kurtpa** [comment]
I've added a 2nd generation attempt in the feature Uri prefix branch that will go out with 0.5.3. Its something I'm going to try improve in a future release

**Comment by kurtpa** [comment]
released with 0.5.3 - this should help the experience however we are bound by the jsonschema2pojo behaviour since that is what we use to generate the pojos
